_pytest/cacheprovider.py

```diff
@@ -14,7 +14,8 @@
 class Cache(object):
     def __init__(self, config):
         self.config = config
-        self._cachedir = config.rootdir.join(".cache")
+        cache_dirname = config.getini("cache_dirname")
+        self._cachedir = config.rootdir.join(cache_dirname)
```

Comment:
I agree, this option is not for setting cache path. It is for change default name of cache directory, wich is created in **rootdir**.
ini option for setting cache path is so more difficult feature.